homeassistant/components/plugwise/sensor.py

```diff
@@ -269,6 +278,7 @@ async def async_setup_entry(
 
     entities: list[PlugwiseSensorEntity] = []
     for device_id, device in coordinator.data.devices.items():
+        await migrate_sensor_entity(hass, coordinator, device_id, device)
```

Comment:
Let's move this into the integration setup instead of the platform and make sure all existing entries are migrated before we set up the platforms.

Reply:
Agreed that we should do them earlier, but it's dependent-on the coordinator data and not just a straightforward 1:1 migration. I.e. we only migrate for the auxiliary physical outdoor temperature sensor, not for all other cases. Using the callback as we do in `gateway.py` for the plugs is more straightforward, but wouldn't work since we wouldn't be able to add additional parameters into the callback (unless I'm mis-understanding the callback definition in `async_migrate_entries`).

Reply:
> but it's dependent-on the coordinator data

The coordinator data is first fetch on integration set up, so, that is not an issue?

Reply:
Yes, I'm trying to see if I can get it working in gateway.py

Reply:
@frenck committed as such (including the async def to def)

Reply:
Working in pw-beta. Will be added to the PR soon...

Reply:
Bouwe's approach is in the most recent commits